tests/amazon_test.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { test, expect } from '@playwright/test';
+import { searchForProduct } from './utils';
+
+test('Search for gift card on Amazon and apply Netflix filter', async ({ page }) => {
+  await page.goto('https://www.amazon.ca');
+
+  await searchForProduct(page, 'gift card');
+
+  await page.waitForSelector('.s-main-slot');
+
+  const resultsSummaryText = await page.innerText('.s-desktop-toolbar');
+
+  const totalResultsMatch = resultsSummaryText.match(/([\d,]+) results/);
+  const totalResults = totalResultsMatch ? parseInt(totalResultsMatch[1].replace(/,/g, ''), 10) : 0;
+
+  console.log(`Number of results before filtering: ${totalResults}`);
+
+  await page.getByLabel('Birthday').getByRole('link', { name: 'Birthday' }).click();
+  await page.waitForTimeout(2000);
+
+  const filteredResultsSummaryText = await page.innerText('.s-desktop-toolbar');
+  const filteredResultsMatch = filteredResultsSummaryText.match(/([\d,]+) results/);
+  const filteredResults = filteredResultsMatch ? parseInt(filteredResultsMatch[1].replace(/,/g, ''), 10) : 0;
+  console.log(`Number of results after filtering: ${filteredResultsMatch}`);
```

Comment:
nit: for me, this logged `Number of results after filtering: 201 results,201`. Not sure where the duplicate `201` is coming from.

![Screenshot 2024-11-02 at 8 08 40 AM](https://github.com/user-attachments/assets/4262fcaa-04b0-4a21-a376-66ad8508b563)